setup_github.sh

```diff
@@ -0,0 +1,89 @@
+
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+source ./functions_github.sh
```

Comment:
if we name the functions similiar and source the correct "provider" by a flag passed we should be able to reuse the `setup.sh` script or most of it